survey/dashboards.py

```diff
@@ -0,0 +1,1708 @@
+    [Input('stored-data', 'data')]
+)
+def update_section_a(data):
+    return create_section_figure(data, 1)  # Index 1 is releasing_potential
+
+
+@dash_app.callback(
+    Output('section-breakdown-b', 'figure'),
+    [Input('stored-data', 'data')]
+)
+def update_section_b(data):
+    return create_section_figure(data, 2)  # Index 2 is embedding_research
+
+
+@dash_app.callback(
+    Output('section-breakdown-c', 'figure'),
+    [Input('stored-data', 'data')]
+)
+def update_section_c(data):
+    return create_section_figure(data, 3)  # Index 3 is linkages_leadership
+
+
+@dash_app.callback(
+    Output('section-breakdown-d', 'figure'),
+    [Input('stored-data', 'data')]
+)
+def update_section_d(data):
+    return create_section_figure(data, 4)  # Index 4 is inclusive_research
+
+
+@dash_app.callback(
+    Output('section-breakdown-e', 'figure'),
+    [Input('stored-data', 'data')]
+)
+def update_section_e(data):
+    return create_section_figure(data, 5)  # Index 5 is digital_capability
+
+
+def create_demographic_chart(data, question_label, sort_by_value=False):
+    if not data or 'survey_responses' not in data:
+        return {}
+
+    question = get_demographic_question(question_label)
+    if not question:
+        print(f"Question not found: {question_label}")
+        return {}
+
+    question_idx = None
+    for idx, field in enumerate(DEMOGRAPHIC_CONFIG['sections'][0]['fields']):
+        if field['label'] == question_label:
+            question_idx = idx
+            break
+
+    if question_idx is None:
+        return {}
+
+    response_counts = defaultdict(int)
+
+    for response in data['survey_responses']:
+        if 'fields' in response and 'answers' in response['fields']:
+            try:
+                answer = response['fields']['answers'][6][question_idx].title()
+                response_counts[answer] += 1
+            except (IndexError, KeyError):
+                continue
+
+    items = list(response_counts.items())
+    if sort_by_value:
+        items.sort(key=lambda x: x[1], reverse=True)
+
+    labels, values = zip(*items) if items else ([], [])
+
+    colors = colour_palette[:len(labels)]
+
+    fig = go.Figure(data=[go.Pie(
+        labels=labels,
+        values=values,
+        hole=0,
+        marker=dict(
+            colors=colors
+        ),
+        textinfo='value+percent',
+        textfont=dict(
+            family=THEME['font_family'],
+            size=13
+        ),
+        textposition='auto',
+    )])
+
+    fig.update_layout(
+        height=400,
+        margin=dict(l=20, r=20, t=40, b=20),
+        paper_bgcolor='white',
+        plot_bgcolor='white',
+        font=dict(
+            family=THEME['font_family'],
+            size=15
+        ),
+        showlegend=True,
+        legend=dict(
+            orientation="h",
+            yanchor="bottom",
+            y=-0.6,
+            xanchor="center",
+            x=0.5,
+        )
+    )
+
+    return fig
+
+
+@dash_app.callback(
```

Comment:
There must be a better way of listing these callbacks but let's discuss it after the user review.